### PR TITLE
Move bundle version to idData

### PR DESCRIPTION
**Pull request by vanruch**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/services/bundle_builder.js [hunk 1]
```diff
@@ -51,30 +55,39 @@ export default class BundleBuilder {
...
-  async extractBundleDataNecessaryForValidationFromStream(readableStream, bundleVersion) {
+  async validateStreamedBundle(readStream, writeStream, bundleItemsCountLimit) {
+    readStream.on('error', (err) => {
+      writeStream.abort(err);
+    });
+    const [minimalBundleForLatestVersionValidation] = await Promise.all([
+      this.extractBundleDataNecessaryForValidationFromStream(readStream),
+      pipeline(readStream, writeStream)
+    ]);
+    if (!this.supportDeprecatedBundleVersions) {
```
Comment (spherefoundry) [inline src/services/bundle_builder.js]: This means that even for version 3 bundles, if this config flag is set, then it will always be validated as pre-3. Instead, you should first try to get the version (assume 1 if not pressent) and then only validate against pre-3 if the flag is set.

File: src/services/bundle_builder.js [hunk 2]
```diff
@@ -51,30 +55,41 @@ export default class BundleBuilder {
...
+    const [minimalBundleForLatestVersionValidation] = await Promise.all([
+      this.extractBundleDataNecessaryForValidationFromStream(readStream),
+      pipeline(readStream, writeStream)
+    ]);
+    if (minimalBundleForLatestVersionValidation.content.idData.version === LATEST_BUNDLE_VERSION) {
+      this.validateBundle(minimalBundleForLatestVersionValidation, bundleItemsCountLimit);
+    } else if (this.supportDeprecatedBundleVersions) {
+      this.validateBundleWithVersionBefore3(minimalBundleForLatestVersionValidation, bundleItemsCountLimit);
+    } else {
+      throw new ValidationError(`Only supported bundle version is: ${LATEST_BUNDLE_VERSION}`);
```
Comment (spherefoundry) [inline src/services/bundle_builder.js]:
```suggestion
      throw new ValidationError(`Only bundles with version ${LATEST_BUNDLE_VERSION} are supported`);
```
